src/resources/parser/texture/legacy-dds.js

```diff
@@ -0,0 +1,159 @@
+                    height: 4,
+                    format: pc.PIXELFORMAT_R8_G8_B8
+                });
+                texture.name = 'dds-legacy-empty';
+                return texture;
+            }
+
+            texture = new pc.Texture(device, {
+                name: url,
+                // #ifdef PROFILER
+                profilerHint: pc.TEXHINT_ASSET,
+                // #endif
+                addressU: isCubemap ? pc.ADDRESS_CLAMP_TO_EDGE : pc.ADDRESS_REPEAT,
+                addressV: isCubemap ? pc.ADDRESS_CLAMP_TO_EDGE : pc.ADDRESS_REPEAT,
+                width: width,
+                height: height,
+                format: format,
+                cubemap: isCubemap
+            });
+
+            var offset = 128;
+            var faces = isCubemap ? 6 : 1;
+            var mipSize;
+            var DXT_BLOCK_WIDTH = 4;
+            var DXT_BLOCK_HEIGHT = 4;
+            var blockSize = fcc === FCC_DXT1 ? 8 : 16;
+            var numBlocksAcross, numBlocksDown, numBlocks;
+            for (var face = 0; face < faces; face++) {
+                var mipWidth = width;
+                var mipHeight = height;
+                for (var i = 0; i < mips; i++) {
+                    if (compressed) {
+                        if (etc1) {
+                            mipSize = Math.floor((mipWidth + 3) / 4) * Math.floor((mipHeight + 3) / 4) * 8;
+                        } else if (pvrtc2) {
+                            mipSize = Math.max(mipWidth, 16) * Math.max(mipHeight, 8) / 4;
+                        } else if (pvrtc4) {
+                            mipSize = Math.max(mipWidth, 8) * Math.max(mipHeight, 8) / 2;
+                        } else {
+                            numBlocksAcross = Math.floor((mipWidth + DXT_BLOCK_WIDTH - 1) / DXT_BLOCK_WIDTH);
+                            numBlocksDown = Math.floor((mipHeight + DXT_BLOCK_HEIGHT - 1) / DXT_BLOCK_HEIGHT);
+                            numBlocks = numBlocksAcross * numBlocksDown;
+                            mipSize = numBlocks * blockSize;
+                        }
+                    } else {
+                        mipSize = mipWidth * mipHeight * 4;
+                    }
+
+                    var mipBuff = floating ? new Float32Array(data, offset, mipSize) : new Uint8Array(data, offset, mipSize);
+                    if (!isCubemap) {
+                        texture._levels[i] = mipBuff;
+                    } else {
+                        if (!texture._levels[i]) texture._levels[i] = [];
+                        texture._levels[i][face] = mipBuff;
+                    }
+                    offset += floating ? mipSize * 4 : mipSize;
+                    mipWidth = Math.max(mipWidth * 0.5, 1);
+                    mipHeight = Math.max(mipHeight * 0.5, 1);
+                }
+            }
+
+            texture.upload();
+
+            return texture;
+        }
+    });
+
+    return {
+        LegacyDdsParser: LegacyDdsParser
+    };
+}());
```

Comment:
Can we not unify our DDS code. Seems weird we have two loaders.

Reply:
Yes I plan to do this in a followup PR.